app/controllers/users/registrations_controller.rb

```diff
@@ -1,5 +1,11 @@
 module Users
   class RegistrationsController < Devise::RegistrationsController
+
+    def edit
+      authorize current_user if their_own?
```

Comment:
Or maybe it can be just called `edit?` instead of `edit_password?`, I'm not sure if the edit function has also other responsibilities.